collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -124,42 +125,52 @@ public void doScrape() throws Exception {
         }
     }
 
+    private static final ConcurrentHashMap<ObjectName, Map<String, MBeanAttributeInfo>> cache =
```

Comment:
This is also a memory leak.

Reply:
Yes indeed this is possible. So may be a system of purge can be put in place. Even though I think monitored Beans should be stable more or less and the Map content won't change a lot after initialization.